### ISSUE TITLE
[Rust] code following `where` is not syntax highlighted

### ISSUE THREAD

**Issue by PonasKovas** [sublimehq/Packages#2786]
- Sublime Version: Version 3.2.2, Build 3211
- OS Version: Arch Linux

```rust
pub trait Foo {
    fn bar()
    where
        Self: Sized;
}

fn main() {
	
}
```

![image](https://user-images.githubusercontent.com/25486263/114314305-c1e23780-9b02-11eb-9292-fde5d0c7ec74.png)

**Comment by keith-hall** [comment]
Labas Ponai Kovai :)

Thanks for reporting. I believe this can be fixed with a two line change - in the `fn-where` context, adding

```yaml
    - match: (?=;)
      pop: true
```

to the bottom seems to fix it. There's also an open PR which may address this: https://github.com/sublimehq/Packages/pull/2305

```diff
--- Shipped Packages/Rust/Rust.sublime-syntax   2020-08-07 14:54:04
+++ Packages/Rust/Rust.sublime-syntax   2020-06-11 08:48:58
@@ -907,6 +905,8 @@
     - include: type-any-identifier
     - match: ':'
       scope: punctuation.separator.rust
+    - match: (?=;)
+      pop: true
 
   fn-body:
     - meta_scope: meta.function.rust
```

**Comment by deathaxe** [comment]
This issue is fixed by #2305
